src/hooks/useControlledState.ts

```diff
@@ -0,0 +1,40 @@
+/*
```

Comment:
We use this hook to tell when the tabs are controlled by a parent or uncontrolled (they manage their own internal state)